cmd/cluster_kubeconfig.go

```diff
@@ -45,41 +53,153 @@ Example 4: hetzner-kube cluster kubeconfig -n my-cluster -p > my-conf.yaml # pri
 
 		FatalOnError(err)
 
-		printContent, _ := cmd.Flags().GetBool("print")
-		force, _ := cmd.Flags().GetBool("force")
-
-		if printContent {
-			fmt.Println(kubeConfigContent)
+		// get sanitized kubeconfig
+		// we need isSanitized flag to ensure we do not want do a merge if we fail to sanitize config
+		// if it fails we simply print out config as it is
+		isSanitized := false
+		newKubeConfig, err := sanitizeKubeConfig(kubeConfigContent, provider.GetCluster().Name, "hetzner")
+		if err != nil {
+			log.Printf("KubeConfig sanitise process failed, default config will be used instead. Error: %s", err.Error())
 		} else {
-			fmt.Println("create file")
+			kubeConfigContent = newKubeConfig
+			isSanitized = true
+		}
 
-			usr, _ := user.Current()
-			dir := usr.HomeDir
-			path := fmt.Sprintf("%s/.kube", dir)
+		if merge, _ := cmd.Flags().GetBool("merge"); merge && isSanitized {
 
-			if _, err := os.Stat(path); os.IsNotExist(err) {
-				os.MkdirAll(path, 0755)
+			if err = mergeKubeConfig(kubeConfigContent); err != nil {
+				log.Fatalf("During merge we encountered the problem: %s", err.Error())
 			}
+			log.Printf("KubeConfig successfully merged!")
+			return
+		}
 
-			// check if there already is an existing config
-			kubeconfigPath := fmt.Sprintf("%s/config", path)
-			if _, err := os.Stat(kubeconfigPath); !force && err == nil {
-				fmt.Println("There already exists a kubeconfig. Overwrite? (use -f to suppress this question) [yN]:")
-				r := bufio.NewReader(os.Stdin)
-				answer, err := r.ReadString('\n')
-				FatalOnError(err)
-				if !strings.ContainsAny(answer, "yY") {
-					log.Fatalln("aborted")
-				}
-			}
+		if save, _ := cmd.Flags().GetBool("save"); save {
 
-			ioutil.WriteFile(kubeconfigPath, []byte(kubeConfigContent), 0755)
+			targetPath := fmt.Sprintf("%s/.kube/%s.yaml", GetHome(), provider.GetCluster().Name)
+			if target, _ := cmd.Flags().GetString("target"); target != "" {
+				targetPath = target
+			}
+			log.Printf("Saving current config to '%s'", targetPath)
+			doConfigWrite(targetPath, kubeConfigContent)
 
-			fmt.Println("kubeconfig configured")
+			return
 		}
+
+		fmt.Println(kubeConfigContent)
 	},
 }
 
+// Write kubeConfig to destination
+func doConfigWrite(dst string, kubeConfig string) (err error) {
+
+	if _, err := os.Stat(path.Dir(dst)); os.IsNotExist(err) {
+		os.MkdirAll(path.Dir(dst), 0755)
+	}
+	return ioutil.WriteFile(dst, []byte(kubeConfig), 0755)
+}
+
+// Create backup of current kubeCongig
+func doConfigCopyBackUp(src string) (err error) {
```

Comment:
```suggestion
func doConfigCopyBackUp(src string) error {
```

Reply:
this done to not assign error directly with `:=`
for example if you going to use:
```
if source, err := os.Open(src); err != nil {
		return
}
```
`source` will available only inside condition, and we want have access to it outside of condition.
if `source` declared, then use of `:=` will rise error.
if to remove declaration of `err` from function then i need also use condition differently as:
```
source, err := os.Open(src)
if err != nil {
   return
}
```
While both variants completely good, first variant probably looks bit nicer.
But here this is how i do usually. also no need do `return` with exact parameters :)
If still want me to use your suggestion, let me know

Reply:
JM2C: the first variant is a bit concise, but make code more hard to understand. We can declare `err` just before the usage (as we do for `source`) and return the error (eventually decorated with extra info), like:
```go
var source, destination *os.File
var err error
if source, err = os.Open(src); err != nil {
    return fmt.Errorf("unable to get source file: %v", err)
}
```
or use the second proposal:
```go
source, err := os.Open(src)
if err != nil {
    return fmt.Errorf("unable to get source file: %v", err)
}
```
IMHO the second variant is more comprehensible when someone look the code for the first time (new contributors, ppl that try to investigate an issue, the user that write the code after few weeks that write the code..).
Off course is my opinion, maybe a 3rd opinion should be useful ;) cc/ @xetys

Reply:
I like the second more, as I'm not a fan of C style declaring variables in the header and than doing the code later. The second one is far better to read and understand, and an actual goal of go to make it looking like this

Reply:
it depends, as i said, all variants is valid in go. but i can agree , i'm always judging from my 5+ experience in go, and for me personally read some-one code in this scope absolutely transparent. but if majority vote for
```
if err != nil {
    return err
}
```
not a problem :D

Reply:
@mavimo @xetys check my comment later below, maybe we really need change way kubeconfig command work, instead of flag usage to print/save/merge , better to change it to sub-command and do `kubeconfig save|print|merge` and then each of subcommand will have own options if necessary?